module1/homework/make_unique/makeUnique-ut.cpp

```diff
@@ -0,0 +1,90 @@
+
+#include <gtest/gtest.h>
+#include <tuple>
+#include "makeUnique.hpp"
+
+namespace TestConsts {
+constexpr int intValue = 42;
+constexpr double doubleValue = 4.2;
+constexpr bool boolValue = false;
+constexpr int lValueInt = -1;
+constexpr int rValueInt = 1;
+}  // namespace TestConsts
+
+struct TestData {
+    TestData(int f, double s, bool t)
+        : first_(f), second_(s), third_(t) {}
+    int first_ = 4;
+    double second_ = 4.2;
+    bool third_ = false;
+};
+
+struct LRValue {
+    LRValue(int value)
+        : value_(value){};
+    LRValue(LRValue&)
+        : value_(TestConsts::lValueInt){};
+    LRValue(LRValue&&)
+        : value_(TestConsts::rValueInt){};
+
+    int value_;
+};
+
+TEST(makeUnique, shouldCreatePointerToInt) {
+    auto ptr = myMakeUnique::make_unique<int>(TestConsts::intValue);
+
+    ASSERT_EQ(*ptr, TestConsts::intValue);
+}
+
+TEST(makeUnique, shouldCreatePointerToDouble) {
+    auto ptr = myMakeUnique::make_unique<double>(TestConsts::doubleValue);
+
+    ASSERT_EQ(*ptr, TestConsts::doubleValue);
+}
+
+TEST(makeUnique, shouldCreatePointerToIntWithDefaultValue) {
+    auto ptr = myMakeUnique::make_unique<int>();
+
+    ASSERT_EQ(*ptr, int());
+}
+
+TEST(makeUnique, shouldCreatePointerToTuple) {
+    auto ptr = myMakeUnique::make_unique<std::tuple<int, double>>(
+        std::make_tuple(TestConsts::intValue, TestConsts::doubleValue));
+
+    ASSERT_EQ(*ptr, std::make_tuple(TestConsts::intValue, TestConsts::doubleValue));
+}
+
+TEST(makeUnique, shouldCreatePointerToOwnStructure) {
+    auto ptr = myMakeUnique::make_unique<TestData>(TestData(TestConsts::intValue,
+                                                            TestConsts::doubleValue,
+                                                            TestConsts::boolValue));
```

Comment:
It's still only one argument that is passed to make_unique and it is TestData. Please pass 3 arguments and create (not copy or move) TestData inside make_unique. So you have only cases when there is either 1 or 0 parameters passed. I need a proof in tests that your function can take any number of arguments :)
```suggestion
    auto ptr = myMakeUnique::make_unique<TestData>(TestConsts::intValue,
                                                   TestConsts::doubleValue,
                                                   TestConsts::boolValue);
```